File: pdu/python/std_msgs/pdu_cdr_conv_Int16.py
from .pdu_pytype_Int16 import Int16
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_Int16(writer: CdrWriter, src: Int16):
    writer.write_int16(src.data)


def cdr_body_to_py_Int16(reader: CdrReader, dst: Int16):
    dst.data = reader.read_int16()
    return dst


def py_to_cdr_Int16(src: Int16) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Int16(writer, src)
    return writer.bytes()


def cdr_to_py_Int16(cdr_payload) -> Int16:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Int16()
    return cdr_body_to_py_Int16(reader, dst)

File: pdu/javascript/std_msgs/pdu_cdr_conv_Int16.js
import { Int16 } from './pdu_jstype_Int16.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduInt16Converter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Int16 } src
     */
    static to_cdr_body(writer, src) {
        writer.write_int16(src.data);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Int16 } dst
     * @returns { Int16 }
     */
    static cdr_body_to_js(reader, dst) {
        dst.data = reader.read_int16();
        return dst;
    }

    /**
     * @param { Int16 } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Int16 }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Int16());
    }
}
